electrum/gui/kivy/uix/dialogs/tx_dialog.py

```diff
@@ -181,6 +182,31 @@ def update(self):
             self.fee_str = _('unknown')
             self.feerate_str = _('unknown')
         self.ids.output_list.update(self.tx.outputs())
+
+        def text_format(addr):
+            """
+            Chooses the appropriate text color and background color to 
+            mark receiving, change and billing addresses.
+
+            Returns: color, background_color
+            """
+            
+            # modified colors (textcolor, background_color) from electrum/gui/qt/util.py
+            GREEN = ("#000000", "#8af296")
```

Comment:
In order not to duplicate the color definitions, it might be cleaner to:

1. Move the color definitions from `electrum.gui.qt.util` to `electrum.util`
2. Make ColorScheme in `electrum.gui.qt.util` extend the one in `electrum.util`.
3. Reference the definition in `electrum.util` here.

Reply:
I completely agree that it would be cleaner to have a central place for colors. I didn't do it for the following reasons:

1. Since this is my first contribution I am not confident to make changes across multiple guis.
2. The color in https://github.com/spesmilo/electrum/blob/master/electrum/gui/qt/util.py#L766  contains 2 colors. A fg_color and bg_color. However I need to also specify the text_color in 
and https://github.com/spesmilo/electrum/pull/5774/files#diff-773ff6da8745ec9ae8fec1cbf69f196f .  So essentially the information in electrum.gui.qt.util is not enough, and would need to be extended.

Reply:
You are right, I initially assumed that the Qt GUI prints foreground colored text on background color. However, it uses the default text color instead. So you are totally right about point 2.

I also agree with you on point 3 but I have to do some nit-picking here. Now that I look at it, the color-choosing mechanism in https://github.com/spesmilo/electrum/commit/5c2beb4ab5d2ec604b4f69878e6be8a4d91944f9#diff-a36dff186a5eff20d9f84c1572ea0acbR605 is quite hard to read but the colors you are using here are only returned as a background color when using a dark theme (like the one kivy uses). In that case, the default text color is assumed to be white(-ish).

![image](https://user-images.githubusercontent.com/598790/69270768-ec60e700-0bd3-11ea-84ad-781c218308a7.png)

In my opinion, while still readable, a black text provides less contrast than a white text on all of these backgrounds. Thus my suggestion would be to use a white text in all cases. What do you think?

Reply:
The display with the color bars is interesting. However the colors seem to be off:
The background green #8af296 is:
![Screenshot_20191120_205754](https://user-images.githubusercontent.com/1935665/69273381-8d51a100-0bd8-11ea-869a-94a35ce106f5.png)
Such that the coloring (electrum testnet, kivy on linux) looks like:
![Screenshot_20191120_205953](https://user-images.githubusercontent.com/1935665/69273506-c68a1100-0bd8-11ea-8da8-afad87fb2e03.png)

With white it would look like:
![Screenshot_20191120_210143](https://user-images.githubusercontent.com/1935665/69273636-fafdcd00-0bd8-11ea-9af9-f2c627cf5d4d.png)

Reply:
Well, I still think the way to go would be to use the darker version of each color with white text 😊 The Qt UI could be changed accordingly.

But of course both could also be changed later since this might be out of scope of your PR.

Does anyone else have an opinion on this?

Reply:
The darker bg colors and white text might look good too; but this looks good too already. I think it's fine as is.